lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -74,4 +76,10 @@ defmodule Mix.Tasks.Compile.Elixir do
     dest = Mix.Project.compile_path
     Mix.Compilers.Elixir.clean(manifest(), dest)
   end
+
+  defp assert_valid_elixirc_paths(paths) do
+    unless is_list(paths) do
+      Mix.raise ":elixirc_paths should be a list of paths, got: #{inspect(paths)}"
```

Comment:
I thought about that but if we get here I think users had to explicitly use `:elixirc_paths` so they likely know where it comes from?